src/lib/homestore_backend/tests/hs_blob_tests.cpp

```diff
@@ -139,71 +139,148 @@ TEST_F(HomeObjectFixture, BasicPutGetDelBlobWRestart) {
-    // }
-    //
-    // // Put blob for all shards in all pg's.
-    // put_blobs(pg_shard_id_vec, num_blobs_per_shard, pg_blob_id);
-    //
-    // PG* pg1;
-    // {
-    //     auto lg = std::shared_lock(_obj_inst->_pg_lock);
-    //     auto iter = _obj_inst->_pg_map.find(pg_id);
-    //     ASSERT_TRUE(iter != _obj_inst->_pg_map.end());
-    //     pg1 = iter->second.get();
-    // }
-    //
-    // auto pg1_iter =
-    //     std::make_shared< homeobject::HSHomeObject::PGBlobIterator >(*_obj_inst, pg1->pg_info_.replica_set_uuid);
-    // ASSERT_EQ(pg1_iter->end_of_scan(), false);
-    //
-    // // Verify PG shard meta data.
-    // sisl::io_blob_safe meta_blob;
-    // pg1_iter->create_pg_shard_snapshot_data(meta_blob);
-    // ASSERT_TRUE(meta_blob.size() > 0);
-    //
-    // auto pg_req = GetSizePrefixedResyncPGShardInfo(meta_blob.bytes());
-    // ASSERT_EQ(pg_req->pg()->pg_id(), pg1->pg_info_.id);
-    // auto u1 = pg_req->pg()->replica_set_uuid();
-    // auto u2 = pg1->pg_info_.replica_set_uuid;
-    // ASSERT_EQ(std::string(u1->begin(), u1->end()), std::string(u2.begin(), u2.end()));
-    //
-    // // Verify get blobs for pg.
-    // uint64_t max_num_blobs_in_batch = 3, max_batch_size_bytes = 128 * Mi;
-    // std::vector< HSHomeObject::BlobInfoData > blob_data_vec;
-    // while (!pg1_iter->end_of_scan()) {
-    //     std::vector< HSHomeObject::BlobInfoData > vec;
-    //     bool end_of_shard;
-    //     auto result = pg1_iter->get_next_blobs(max_num_blobs_in_batch, max_batch_size_bytes, vec, end_of_shard);
-    //     ASSERT_EQ(result, 0);
-    //     for (auto& v : vec) {
-    //         blob_data_vec.push_back(std::move(v));
-    //     }
-    // }
-    //
-    // ASSERT_EQ(blob_data_vec.size(), num_shards_per_pg * num_blobs_per_shard);
-    // for (auto& b : blob_data_vec) {
-    //     auto g = _obj_inst->blob_manager()->get(b.shard_id, b.blob_id, 0, 0).get();
-    //     ASSERT_TRUE(!!g);
-    //     auto result = std::move(g.value());
-    //     LOGINFO("Get blob pg {} shard {} blob {} len {} data {}", 1, b.shard_id, b.blob_id, b.blob.body.size(),
-    //             hex_bytes(result.body.cbytes(), 5));
-    //     EXPECT_EQ(result.body.size(), b.blob.body.size());
-    //     EXPECT_EQ(std::memcmp(result.body.bytes(), b.blob.body.cbytes(), result.body.size()), 0);
-    //     EXPECT_EQ(result.user_key.size(), b.blob.user_key.size());
-    //     EXPECT_EQ(result.user_key, b.blob.user_key);
-    //     EXPECT_EQ(result.object_off, b.blob.object_off);
-    // }
+    // Generate test data
+    uint64_t num_shards_per_pg = 3;
+    uint64_t num_blobs_per_shard = 5;
+    std::map< pg_id_t, std::vector< shard_id_t > > pg_shard_id_vec;
+    std::map< pg_id_t, blob_id_t > pg_blob_id;
+
+    pg_id_t pg_id{1};
+    create_pg(pg_id);
+    for (uint64_t i = 0; i < num_shards_per_pg; i++) {
+        auto shard = create_shard(1, 64 * Mi);
+        pg_shard_id_vec[1].emplace_back(shard.id);
```

Comment:
Suggest making `pg_id` constexpr and refer to it wherever needed in the testcase, rather than using `1` directly.

Reply:
Fixed.